rust/processor/src/models/token_v2_models/v2_token_datas.rs

```diff
@@ -139,13 +140,78 @@ impl TokenDataV2 {
                     last_transaction_version: txn_version,
                     last_transaction_timestamp: txn_timestamp,
                     decimals: None,
+                    is_deleted_v2: Some(false),
                 },
             )))
         } else {
             Ok(None)
         }
     }
 
+    /// This handles the case where token is burned but objectCore is still there
+    pub async fn get_burned_nft_v2_from_write_resource(
```

Comment:
isn't this case already covered by `get_burned_nft_v2_from_delete_resource` below?

Reply:
no. this is write resource.

Reply:
when a token gets burnt won't it always have a delete resource?

Reply:
What if it's a partial burn and there are resources left?

Reply:
won't the token resource get deleted so there will always be a delete resource?

Reply:
no that's not how it works. unless the resource group is entirely destroyed it'll emit a write resource event :(